projects/eval-adxl355-pmdz/src/platform/maxim/main.c

```diff
@@ -61,16 +51,11 @@ int main()
 
 	adxl355_ip.comm_init.spi_init = adxl355_spi_ip;
 
-#ifdef IIO_EXAMPLE
-	ret = iio_example_main();
-#endif
-
-#ifdef IIO_TRIGGER_EXAMPLE
-	struct no_os_gpio_desc *adxl355_gpio_desc;
 	struct no_os_irq_ctrl_desc *nvic_desc;
 	struct no_os_irq_init_param nvic_ip = {
 		.platform_ops = &max_irq_ops,
 	};
+	struct no_os_gpio_desc *adxl355_gpio_desc;
 
 	/* Initialize DATA READY pin */
 	ret = no_os_gpio_get_optional(&adxl355_gpio_desc, &adxl355_gpio_drdy_ip);
```

Comment:
this DRDY pin is really weird, it's initialized for the iio_trigger_example but it's not used anywhere, so if anyone has an explanation for this let me know

Reply:
@CiprianRegus came with a good explanation, on stm32, EXTI is automatically configured as GPIO input by the HAL but on aducm3029 and maxim, it's not, so this extra step is needed. In the modified code, this is kept so I don't think it will break anything.